sim/firesim-lib/src/main/scala/passes/AsyncResetReg.scala

```diff
@@ -54,7 +55,18 @@ object AsyncResetRegPass extends firrtl.passes.Pass {
       case _ => m
     } 
 
-  def run(c: Circuit): Circuit = {
-    c.copy(modules = c.modules map transform)
+  def execute(cs: CircuitState): CircuitState = {
+    val transformedMods = cs.circuit.modules.map(transform)
+
+    // Drop AsyncResetReg BlackBoxSourceHelper annotations
```

Comment:
Is this just to be more complete?

Reply:
Otherwise the AsyncResetReg ends up in the list of blackbox resource files. Since the pass deletes the module, it should delete the annotation, or the blackbox Verilog definition can conflict with the pure-FIRRTL mock synchronous implementation.

Reply:
Got it.